File: ios-kyrie-ai/KyrieAI/Onboarding/OnboardingDraft.swift
//
//  OnboardingDraft.swift
//  KyrieAI
//
//  Mutable state collected across the onboarding steps.
//

import SwiftUI

@Observable
final class OnboardingDraft {
    var name: String = ""
    var age: Int = 16
    var heightInches: Int = 70
    var skillLevel: SkillLevel? = nil
    var position: Position? = nil
    var dominantHand: Hand? = nil
    var goals: Set<TrainingGoal> = []
    var trainingDays: Set<Weekday> = []
    var specificRequests: String = ""

    func canContinue(from step: OnboardingStep) -> Bool {
        switch step {
        case .welcome: return true
        case .drill: return true
        case .progress: return true
        case .stat: return true
        case .impact: return true
        case .turnovers: return true
        case .rate: return true
        case .quizIntro: return true
        case .name: return !name.trimmingCharacters(in: .whitespaces).isEmpty
        case .physicals: return true
        case .skill: return skillLevel != nil
        case .position: return position != nil
        case .hand: return dominantHand != nil
        case .goals: return !goals.isEmpty
        case .availability: return !trainingDays.isEmpty
        case .requests: return true
        case .ready: return true
        }
    }
}

enum OnboardingStep: Int, CaseIterable {
    case welcome, drill, progress, stat, impact, turnovers, rate, quizIntro, name, physicals, skill, position, hand, goals, availability, requests, ready

    var progressIndex: Int { rawValue }
}

File: android/app/src/main/java/com/rork/kyrieai/data/OnboardingDraft.kt
package com.rork.kyrieai.data

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.compose.runtime.mutableStateListOf

/** Mutable state collected across the onboarding steps. */
class OnboardingDraft {
    var name by mutableStateOf("")
    var age by mutableStateOf(16)
    var heightInches by mutableStateOf(70)
    var skillLevel by mutableStateOf<SkillLevel?>(null)
    var position by mutableStateOf<Position?>(null)
    var dominantHand by mutableStateOf<Hand?>(null)
    val goals = mutableStateListOf<TrainingGoal>()
    val trainingDays = mutableStateListOf<Weekday>()
    var specificRequests by mutableStateOf("")

    fun canContinue(step: OnboardingStep): Boolean = when (step) {
        OnboardingStep.WELCOME -> true
        OnboardingStep.NAME -> name.trim().isNotEmpty()
        OnboardingStep.PHYSICALS -> true
        OnboardingStep.SKILL -> skillLevel != null
        OnboardingStep.POSITION -> position != null
        OnboardingStep.HAND -> dominantHand != null
        OnboardingStep.GOALS -> goals.isNotEmpty()
        OnboardingStep.AVAILABILITY -> trainingDays.isNotEmpty()
        OnboardingStep.REQUESTS -> true
        OnboardingStep.READY -> true
        OnboardingStep.RATE -> true
    }
}

/** Order matches the iOS flow: welcome → details → ready → rate. */
enum class OnboardingStep {
    WELCOME, NAME, PHYSICALS, SKILL, POSITION, HAND, GOALS, AVAILABILITY, REQUESTS, READY, RATE;

    val progressIndex: Int get() = ordinal
}
